contracts/votes/src/types.rs

```diff
@@ -121,6 +132,12 @@ impl Proposal {
         voter: Address,
         asset_id: Address,
     ) -> i128 {
+        // Check if voter has already voted and has the same vote.
+        let vote_key = VotingHistory::Voting(voter.clone(), proposal_id);
+        let has_key = env.storage().instance().has(&vote_key);
+        if has_key && in_favor == env.storage().instance().get::<VotingHistory, bool>(&vote_key).unwrap() {
```

Comment:
@deepink-pat i still don't get this. Why is the error only triggered if in favour?

Reply:
`in_favor` compares the inside `env.storage().instance().get::<VotingHistory, bool>(&vote_key).unwrap()`. If they are equal. Then it means that the user is trying to vote again with the same vote.

Reply:
`in_favor == env.storage().instance().get::<VotingHistory, bool>(&vote_key).unwrap()`